kernel/kselftest.py

```diff
@@ -153,6 +160,63 @@ def test(self):
         if self.error:
             self.fail("Testcase failed during selftests")
 
+    def kselftestengine(self, arg_payload):
+        """
+        Run the kself test case command
+        Ex:./ksm_tests -H -s 100
+        """
+        cmd_output = subprocess.Popen(arg_payload,
+                                      stdout=subprocess.PIPE,
+                                      stderr=subprocess.STDOUT)
+        output, error = cmd_output.communicate()
+        output_payload = output.decode("utf-8")
+        self.log.info("======{} Test Result======".format(arg_payload))
+        self.log.info(output_payload)
+
+    def test_kself(self):
+        """
+        Supported <test type>:
+        -M (page merging)
+        -Z (zero pages merging)
+        -N (merging of pages in different NUMA nodes)
+        -U (page unmerging)
+        -P evaluate merging time and speed.
+        For this test, the size of duplicated memory area (in MiB)
+        must be provided using -s option
+        -H evaluate merging time and speed of area allocated mostly with
+        huge pages For this test, the size of duplicated memory area(in MiB)
+        must be provided using -s option
+        -C evaluate the time required to break COW of merged pages.
+        """
+        ksm_test_dir = self.sourcedir + "/" + "vm" + "/"
+        self.test_list = ["-M", "-Z", "-N", "-U", "-C"]
+        if os.path.exists(ksm_test_dir):
+            os.chdir(ksm_test_dir)
+            if(self.test_type == "-H" or self.test_type == "-P"):
+                arg_payload = [r"./ksm_tests", self.test_type,
+                               self.Size_flag, self.Dup_MM_Area]
+                try:
+                    self.kselftestengine(arg_payload)
+                except SystemError:
+                    self.fail("Test failed ./ksm_tests{} {} {} cmd"
+                              .format(
+                                  self.test_type,
+                                  self.Size_flag,
+                                  self.Dup_MM_Area))
```

Comment:
can be more simple than this, as below, for easy indentation too.

`self.fail("Test failed {}" .format(" ".join(arg_payload))`